packages/multichain-transactions-controller/src/MultichainTransactionsController.ts

```diff
@@ -404,45 +400,64 @@ export class MultichainTransactionsController extends BaseController<
 
     Object.entries(transactionsUpdate.transactions).forEach(
       ([accountId, newTransactions]) => {
-        // Account might not have any transactions yet, so use `[]` in that case.
-        const oldTransactions =
-          this.state.nonEvmTransactions[accountId]?.transactions ?? [];
+        updatedTransactions[accountId] = {};
 
-        const filteredNewTransactions =
-          this.#filterTransactions(newTransactions);
+        newTransactions.forEach((tx) => {
+          const chain = tx.chain as string;
```

Comment:
Same here, should already be a `CaipChainId`.

```suggestion
          const { chain } = tx;
```